src/core/tls.go

```diff
@@ -77,8 +77,8 @@ func (t *tcptls) init(tcp *tcp) {
 	}
 }
 
-func (t *tcptls) configForOptions(options *tcpOptions) *tls.Config {
-	config := *t.config
+func (t *tcptls) configForOptions(options *tcpOptions, serverName string) *tls.Config {
```

Comment:
If the `tcpOptions` contain `tlsSNI`, then I don't see a reason to pass `serverName` here... we could just make sure the field in the `tcpOptions` is correct, and read it from there (it looks like that's effectively what already happens, it just passes through the arguments as a needless extra step).